File: src/main/java/com/objectbrains/sti/constants/EmailAddressType.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package com.objectbrains.sti.constants;

/**
 *
 * @author jaimel
 */
public enum EmailAddressType {
    
    PRIMARY_EMAIL(1, "Primary Email"),
    NON_PRIMARY_EMAIL(2, "Non-Primary Email"),
    BUSINESS_EMAIL(3, "Business Email");
    
    private final int emailAddressType;
    private final String emailAddressDesc;
    
    private EmailAddressType(int type, String desc) {
        this.emailAddressType = type;
        this.emailAddressDesc = desc;
    }

    public int getEmailAddressType() {
        return emailAddressType;
    }

    public String getEmailAddressDesc() {
        return emailAddressDesc;
    }
       
    public static EmailAddressType getEmailAddressType(int id) {
        for (EmailAddressType eat : EmailAddressType.values()) {
            if (eat.getEmailAddressType()== id) {
                return eat;
            }
        }   
        return null;
    }
}

File: web-src/src/app/shared-modules/main/enums/EmailAddressType.ts
export enum EmailAddressType {
    PRIMARY_EMAIL = 1,
    NON_PRIMARY_EMAIL = 2,
    BUSINESS_EMAIL = 3
}

export function getEmailAddressTypeLabel(source: EmailAddressType): string {
    switch (source) {
        case EmailAddressType.PRIMARY_EMAIL:
            return "Primary Email";
        case EmailAddressType.NON_PRIMARY_EMAIL:
            return "Non-Primary Email";
        case EmailAddressType.BUSINESS_EMAIL:
            return "Business Email"
        default:
            return "Unknown";
    }
}

export const EmailAddressTypeKeyValueArray = [
    {id: EmailAddressType.PRIMARY_EMAIL, label: getEmailAddressTypeLabel(EmailAddressType.PRIMARY_EMAIL)},
    {id: EmailAddressType.NON_PRIMARY_EMAIL, label: getEmailAddressTypeLabel(EmailAddressType.NON_PRIMARY_EMAIL)},
    {id: EmailAddressType.BUSINESS_EMAIL, label: getEmailAddressTypeLabel(EmailAddressType.BUSINESS_EMAIL)}
];